tests/integration/ceph_object_test.go

```diff
@@ -0,0 +1,103 @@
+/*
+Copyright 2021 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package integration
+
+import (
+	"testing"
+
+	"github.com/rook/rook/tests/framework/clients"
+	"github.com/rook/rook/tests/framework/installer"
+	"github.com/rook/rook/tests/framework/utils"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+func TestCephObjectSuite(t *testing.T) {
+	if installer.SkipTestSuite(installer.CephTestSuite) {
+		t.Skip()
+	}
+
+	// Skip the suite if CSI is not supported
+	kh, err := utils.CreateK8sHelper(func() *testing.T { return t })
+	require.NoError(t, err)
+	checkSkipCSITest(t, kh)
+
+	s := new(ObjectSuite)
+	defer func(s *ObjectSuite) {
+		HandlePanics(recover(), s.TearDownSuite, s.T)
+	}(s)
+	suite.Run(t, s)
+}
+
+type ObjectSuite struct {
+	suite.Suite
+	helper    *clients.TestClient
+	settings  *installer.TestCephSettings
+	installer *installer.CephInstaller
+	k8sh      *utils.K8sHelper
+}
+
+func (s *ObjectSuite) SetupSuite() {
+	namespace := "object-ns"
+	s.settings = &installer.TestCephSettings{
+		ClusterName:               "object-cluster",
+		Namespace:                 namespace,
+		OperatorNamespace:         installer.SystemNamespace(namespace),
+		StorageClassName:          installer.StorageClassName(),
+		UseHelm:                   false,
+		UsePVC:                    installer.UsePVC(),
+		Mons:                      3,
+		SkipOSDCreation:           false,
+		UseCSI:                    true,
+		EnableAdmissionController: true,
+		UseCrashPruner:            true,
+		RookVersion:               installer.VersionMaster,
+		CephVersion:               installer.PacificVersion,
+	}
+	s.settings.ApplyEnvVars()
+	s.installer, s.k8sh = StartTestCluster(s.T, s.settings, objectSuiteMinimalTestVersion)
+	if s.k8sh.VersionAtLeast("v1.16.0") {
+		s.settings.EnableVolumeReplication = true
+	}
+	s.helper = clients.CreateTestClient(s.k8sh, s.installer.Manifests)
+}
+
+func (s *ObjectSuite) AfterTest(suiteName, testName string) {
+	s.installer.CollectOperatorLog(suiteName, testName)
+}
+
+func (s *ObjectSuite) TearDownSuite() {
+	s.installer.UninstallRook()
+}
+
+func (s *ObjectSuite) TestWithTLS() {
+	if utils.IsPlatformOpenShift() {
+		s.T().Skip("object store tests skipped on openshift")
+	}
+
+	tls := true
+	runObjectE2ETest(s.helper, s.k8sh, s.Suite, s.settings.Namespace, tls)
```

Comment:
What if we move the method `runObjectE2ETest()` into this go file? I wouldn't expect any other suite to be calling it now. All the other tests are calling `runObjectE2ETestLite()`, right?

Reply:
I did this. The `ceph_base_object_test.go` is now hilariously small. Maybe I moved too much

Reply:
Looks right to me!